docs/docs/user-guide/using-cli-context.md

```diff
@@ -0,0 +1,107 @@
+# Use CLI for Microsoft 365 context
+
+CLI for Microsoft 365 context may store any kind of option and its value. 
+
+## How does it work
+
+When a command is executed CLI will first check for the `.m365rc.json` file in the current directory (this is where the context is saved once you set any option). If present, CLI will parse the context object and check if any of the options defined in it may be used for the currently executed command. If yes CLI will execute this command with the option and its value taken from the context. 
+
+For example if we have the following context defined in the `.m365rc.json` file
+```json
+{
+  "context": {
+    "groupName": "test group",
+    "listTitle": "test list"
+  }
+}
+```
+
+And we will execute the following command: 
+```powershell
+m365 spo listitem list --webUrl "https://contoso.sharepoint.com/sites/sample"
+```
+
+The command is missing the `listTitle` required option but instead of failing it will be executed as this option is present in the current context. The `groupName` option will not be used from the context as the `spo listitem list` command does not have it.
+
+When the same option is defined in the context and also in the command itself then the value defined in the command will be used.
+
+## Guidance
```

Comment:
What you describe here is how to get started using the context. I think it belongs above the current 'How does it work'  section. I think that's easier on the user to follow along.

So, from the top:
1. What is a context and why would you use it
2. How to get started using the context
3. How does it work

Reply:
ok. I updated the order and the title of the paragraphs